tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -453,29 +499,30 @@ static void fastDiscardSpan() {
      * <p>Does not construct the Span object if no subscriber will see it.
      */
     public static void fastCompleteSpan() {
-        fastCompleteSpan(Collections.emptyMap());
+        fastCompleteSpan(NoTagRecorder.INSTANCE, NoTagRecorder.INSTANCE);
     }
 
     /**
      * Like {@link #fastCompleteSpan()}, but adds {@code metadata} to the current span being completed.
-     *
-     * @deprecated Use {@link #fastCompleteSpan()}
      */
-    @Deprecated
     public static void fastCompleteSpan(Map<String, String> metadata) {
```

Comment:
I don't want to support nulls in the API/contract, but I prefer for observability code to be tolerant to null inputs. I think adding the `recorder.isEmpty(data)` check in the Span.Builder adapter would allow this to accept nulls, otherwise the Span object will NPE.